code/modules/mob/living/say.dm

```diff
@@ -252,7 +257,12 @@ var/list/department_radio_keys = list(
 
 	var/rendered = render_speech(speech)
 
-	for (var/atom/movable/listener in listeners)
+	var/list/listening_nonmobs = listeners.Copy()
```

Comment:
This is unnecessary, the for loop below creates a copy of `listeners` internally and you can safely remove entries from it

Reply:
There's a second for loop that just wants the non-mobs though. That loop needs to come *after* this loop so it's not internal anymore.

Reply:
That's fine, the for loop iterates over the copy but the code inside the loop removes elements from the actual `listeners`

Reply:
We still need listeners to have mobs included in it for when we send_speech_bubble though

Reply:
Oh, I totally missed that, my bad.